feat: enable single component mode on push task

lcarva opened the pull request:
## Describe your changes

Single component mode allows only releasing of a single image from the snapshot - the image that caused the snapshot to be created in the first place. This is particularly useful for mono repos.

This commit adds a step to the push-snapshot-to-quay task to implement the single component mode by reducing the snapshot to a single image if the mode is enabled.

## Checklist before requesting a review
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)

lcarva committed:
feat: enable single component mode on push task

Single component mode allows only releasing of a single image from the
snapshot - the image that caused the snapshot to be created in the first
place. This is particularly useful for mono repos.

This commit adds a step to the push-snapshot-to-quay task to implement
the single component mode by reducing the snapshot to a single image if
the mode is enabled.

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

Paul123111 commented:
To keep the sample tasks/pipeline simple, we won't add any more features to `push-snapshot-to-quay`, and are instead recommending for people to create a duplicate of this task/pipeline and contribute to that instead.

lcarva commented:
@Paul123111, are you saying I should add another Task to this repo, or telling me to maintain a fork of that Task on my own repo?

Paul123111 commented:
You should add another task to this repo

lcarva committed:
feat: add push-snapshot-to-quay-single-component-mode task

This is copy of the push-snapshot-to-quay task but with support for
single component mode.

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

lcarva commented:
Updated the PR to create a copy of the task. To be honest, this doesn't seem like the best way to collaborate on these community tasks. This PR is now 4 times larger. Most of the content is completely duplicated. Fixes on one task will not automatically propagate to the other. Collaboration wise, this is not much better than if I were to maintain my own fork.

I can imagine how adding a new feature adds to what you have to maintain (I see you are the single OWNER of that task). If it helps, I'm happy to maintain the existing Task with you so we don't have to duplicate efforts. wdyt?

lcarva commented:
I don't want to maintain a copy of the Task. IMO that's not the right way forward and it's counter to collaboration. I'm open to sharing maintenance of the existing Task and contributing to it in the future.

For now, I'll close this out and simply maintain my own fork. :+1:

Paul123111 commented:
I know you closed this and were going to maintain your own fork, but `push-snapshot-to-quay` has been changed to allow contributions, and there is a new `sample-pipeline` that's serving as an example in the place of `push-snapshot-to-quay` and `notify-failure-on-slack`, so you can add your changes directly to the task/pipeline instead of having to maintain a fork.

Thanks for your concern about the sample tasks and for volunteering to help maintain this task, I think this task should be easier for users to contribute to now.
